src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ServiceProviderEngineScope.cs

```diff
@@ -176,6 +176,12 @@ private void ClearState()
             // try to return to the pool while somebody is trying to access ResolvedServices.
             lock (_scopeLock)
             {
+                // Don't attempt to dispose if we're already disposed
+                if (_state == null)
```

Comment:
I could also check _disposed.

Reply:
Why do we only set _state to null below if Return returns true?  We don't want to always null it out?

Reply:
Super weird edge case around root scopes vs child scopes https://github.com/dotnet/runtime/pull/50463#discussion_r605859770

Reply:
Maybe add the comment to the code, so we remember next time?

Reply:
Note, it also doesn't get nulled out when the pool is full.

Reply:
It's in the comment below this statement.